httpmw/ip_cognito_auth.go

```diff
@@ -112,6 +112,16 @@ func (k *Key) RSA256() (*rsa.PublicKey, error) {
 	return pub, nil
 }
 
+type IpCognitoParams struct {
+	Srv      cognitoidentityprovideriface.CognitoIdentityProviderAPI
+	Cache    redis.Cmdable
+	ClientID string
+	IpRange  string
+	Expire   time.Duration
+	Username string
+	Groups   []string
```

Comment:
I suggest instead of passing username and groups pass an entire user. Something like:
```go
type IpCognitoParams struct {
	Srv      cognitoidentityprovideriface.CognitoIdentityProviderAPI
	Cache    redis.Cmdable
	ClientID string
	IpRange  string
	Expire   time.Duration
	User     *CognitoUser
}
```